java/Oleksii-Riabov/src/com/zoolatech/lecture6/tasks/_3/UserRepository.java

```diff
@@ -0,0 +1,22 @@
+package com.zoolatech.lecture6.tasks._3;
+
+public class UserRepository {
+
+    private final UserCache userCache;
+    private final UserTable userTable;
+
+    public UserRepository(UserCache userCache, UserTable userTable) {
+        this.userCache = userCache;
+        this.userTable = userTable;
+    }
+
+    public String findUserEmail(String id) throws UserMissingException{
+        if (userCache.findUser(id).isPresent()) {
```

Comment:
Try to simplify your code with the help of Optional's `or..` and `map` methods

Reply:
Code has been simplified with the help of Optional's or operators.